### PR TITLE
Add new deserializers and graph functions

### PR DESCRIPTION
**Pull request by JohannesSoots**
- add new graph functions 
- stop client, if can't establish connection
- make poolboy max_overflow configurable
- deserialize Gremlin Path and Map

### CODE REVIEW

File: lib/gremlex/client.ex [hunk 1]
```diff
@@ -53,10 +53,15 @@ defmodule Gremlex.Client do
 
       error ->
         Logger.error("Error establishing connection to server: #{inspect(error)}")
-        GenServer.start_link(__MODULE__, %{}, [])
+        GenServer.start_link(__MODULE__, nil, [])
```
Comment (wildstrings) [inline lib/gremlex/client.ex]: Consider returning `{:stop, "Socket is missing"}` right here to be more explicit. `init()` won't be needed then.

File: lib/gremlex/path.ex [hunk 2]
```diff
@@ -0,0 +1,16 @@
+defmodule Gremlex.Path do
+  alias Gremlex.Path
+  alias Gremlex.Deserializer
+
+  @type t :: %Gremlex.Path{labels: List.t(), objects: List.t()}
```
Comment (wildstrings) [inline lib/gremlex/path.ex]: Can alias be used here as well?

File: lib/gremlex/client.ex (hunk 1 above)
Comment (JohannesSoots) [reply lib/gremlex/client.ex]: [`init/1`](https://hexdocs.pm/elixir/GenServer.html#c:init/1) is required callback for GenServer

Comment (wildstrings) [reply lib/gremlex/client.ex]: OK, it is indeed. Good to go then.

**Pull request merged**